Tools/wasm/wasm_build.py

```diff
@@ -80,24 +87,28 @@
 """
 
 
-def get_emscripten_root(emconfig: pathlib.Path = EM_CONFIG) -> pathlib.PurePath:
-    """Parse EM_CONFIG file and lookup EMSCRIPTEN_ROOT
+def get_emscripten_root(
+    emconfig: pathlib.Path = EM_CONFIG,
+) -> Iterable[pathlib.PurePath]:
+    """Parse EM_CONFIG file and lookup EMSCRIPTEN_ROOT and NODE_JS
 
     The ".emscripten" config file is a Python snippet that uses "EM_CONFIG"
     environment variable. EMSCRIPTEN_ROOT is the "upstream/emscripten"
     subdirectory with tools like "emconfigure".
     """
     if not emconfig.exists():
-        return _MISSING
+        return _MISSING, _MISSING
```

Comment:
Why the double return? Is this just to make it an iterable, or specifically an iterable of 2? If the latter then:

```suggestion
        return [_MISSING]
```

If the latter then should this function be changed to return a tuple of two paths?

Reply:
It's me getting the hang out of typing. You are right, a structured tuple is the better return type here.